src/core/crypto/ecdsa.hpp

```diff
@@ -242,6 +354,27 @@ class P256
         {
             return otPlatCryptoEcdsaVerify(this, &aHash, &aSignature);
         }
+
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+        /**
+         * This method uses the key reference passed to verify the ECDSA signature of a hashed message.
+         *
+         * @param[in] aKeyRef          The reference to the public key to be used
+         * @param[in] aHash            The SHA-256 hash value of a message to use for signature verification.
+         * @param[in] aSignature       The signature value to verify.
+         *
+         * @retval kErrorNone          The signature was verified successfully.
+         * @retval kErrorSecurity      The signature is invalid.
+         * @retval kErrorInvalidArgs   The key or has is invalid.
+         * @retval kErrorNoBufs        Failed to allocate buffer for signature verification
+         *
+         */
+        Error VerifyUsingKeyRef(uint32_t aKeyRef, const Sha256::Hash &aHash, const Signature &aSignature) const
```

Comment:
Where do we use this method?
Do we need this? 
This is in `PublicKey` class which I guess don't need to be saved in PSA?

Reply:
Yes, we do need it. ECDSA can be used by the user for app usecases and it is useful to have verify method along with Sign.

Reply:
Ok, I can add a public facing API to use this API. But I do see a use for this API.

Reply:
The only place we use `Ecdsa::Verify()` today is in `Srp::Server` where we verify the signature of a received "SRP Update" message. In this case, the public key is itself read from the message and won't be saved in the PSA as a key ref (`Srp::Server` only knows the public key which is not security critical).

Reply:
Ok, in this case, can we have a abstraction for verify, but remove it from the core header? This way anyone can use the API to verify, but we dont need to maintain it in the core.